application/api/controller.py

```diff
@@ -0,0 +1,462 @@
+                     'date_created': bucketlist.created_by,
+                     'description': bucketlist.description,
+                     'message': "Bucketlist edited successfully."
+
+                     }
+                ), 200)
+        else:
+            return make_response(
+                jsonify(
+                    {
+                        'message': user_id
+
+                    }
+                ), 401)
+
+    def delete(self, bucketlist_id):
+        token = request.headers.get('Authorization')
+        user_id = User.verify_token(token)
+        if isinstance(user_id, int):
+            bucketlist = BucketList.query.filter_by(
+                id=bucketlist_id, created_by=user_id).first()
+            if bucketlist:
+                db.session.delete(bucketlist)
+                db.session.commit()
+                return make_response(
+                    jsonify(
+                        {
+                            'message': "Bucketlist deleted successfully."
+                            
+                        }
+                    ), 204)
+            else:
+                return make_response(
+                    jsonify(
+                        {
+                            'message': "Bucketlist with given id does not exist."
+                        }
+                    ), 200)
+        else:
+            return make_response(
+                jsonify({
+                    'message': user_id
+
+                }
+                ), 404)
+
+    def get(self, bucketlist_id):
+        token = request.headers.get('Authorization')
+        user_id = User.verify_token(token)
+        if isinstance(user_id, int):
+            bucketlist = BucketList.query.filter_by(id=bucketlist_id).first()
+            item = BucketListItems.query.filter_by(
+                bucketlist_id=bucketlist_id).all()
+            if bucketlist:
+                items = []
+                if item:
+                    for b_item in item:
+                        items.append({"name": b_item.name,
+                                      "id": b_item.id,
+                                      "date_created": b_item.date_created,
+                                      "date_modified": b_item.date_modified,
+                                      "done": b_item.done})
+                    return make_response(
+                        jsonify(
+
+                            {   'id':bucketlist.id,
+                                'name': bucketlist.name,
+                                'description': bucketlist.description,
+                                'items': items,
+                                'message': "Bucketlist obtained successfully.",
+                                'created_by': bucketlist.created_by,
+                                'date_created': bucketlist.date_created
+                            }
+                        ), 200)
+
+                return make_response(
```

Comment:
this code seems like a duplication of line 218-228. If thats the case, this could be refactored